vendors/nxp/boards/lpc54018iotmodule/ports/secure_sockets/iot_secure_sockets.c

```diff
@@ -474,6 +474,204 @@ int32_t SOCKETS_Send( Socket_t xSocket,
+        char * buffLoc = NULL;
+        QCA_CONTEXT_STRUCT * enetCtx = wlan_get_context();
+        A_STATUS xSelectStatus;
+
+        if( ( SOCKETS_INVALID_SOCKET != xSocket ) &&
+            ( NULL != pvBuffer ) &&
+            ( NULL != pxAddress ) )
+        {
+            if( !( nxpsecuresocketsSOCKET_READ_CLOSED_FLAG & pxContext->xShutdownFlags ) )
+            {
+                pxContext->xRecvFlags = ( BaseType_t ) ulFlags;
+
+                for( ; ; )
+                {
+                    /* Check if there is anything to be received on this socket. */
+                    xSelectStatus = ( A_STATUS ) t_select( enetCtx,
+                                                           ( uint32_t ) pxContext->xSocket,
+                                                           nxpsecuresocketsONE_MILLISECOND );
+
+                    if( xSelectStatus == A_OK )
+                    {
+                        lStatus = qcom_recvfrom( ( int ) pxContext->xSocket,
+                                                 &buffLoc,
+                                                 xDataLength,
+                                                 pxContext->xRecvFlags,
+                                                 ( struct sockaddr * ) &xTempAddress,
+                                                 ( socklen_t * ) &xAddressLength );
+
+                        if( lStatus >= 0 )
+                        {
+                            if( buffLoc != NULL )
+                            {
+                                memcpy( pvBuffer, buffLoc, lStatus );
+                                zero_copy_free( buffLoc );
+                            }
+
+                            pxAddress->ulAddress = SOCKETS_htonl( xTempAddress.sin_addr.s_addr );
+                            pxAddress->ucSocketDomain = xTempAddress.sin_family;
+                            pxAddress->ucLength = ( uint8_t ) sizeof( *pxAddress );
+                            pxAddress->usPort = SOCKETS_htons( xTempAddress.sin_port );
+                            *xAddressLength = sizeof( SocketsSockaddr_t );
+                            break;
+                        }
+                        else
+                        {
+                            lStatus = SOCKETS_SOCKET_ERROR;
+                            break;
+                        }
+                    }
+                    else if( xSelectStatus == A_ERROR )
+                    {
+                        if( ( xTaskGetTickCount() - xTimeOnEntering ) < pxContext->ulRecvTimeout )
+                        {
+                            vTaskDelay( nxpsecuresocketsFIVE_MILLISECONDS );
+                        }
+                        else
+                        {
+                            lStatus = SOCKETS_SOCKET_ERROR;
+                            break;
+                        }
+                    }
+                    else
+                    {
+                        lStatus = SOCKETS_SOCKET_ERROR;
```

Comment:
What happens when select does not detect any data on socket?  Do we return error ? I should return equivalent of EWOULDBLOCK to indicate application that it is an error. 
In network abstraction, we can use this feature to implement PEEK option , if not supported by the socket.

Reply:
When select return A_ERROR it means timeout or no activity. Does